### PR TITLE
Support for codepage-based encodings

### PR DESCRIPTION
**Pull request by skolima**

### CODE REVIEW

File: NuKeeper.Inspection.Tests/RepositoryInspection/ProjectFileReaderTests.cs [hunk 1]
```diff
@@ -86,6 +87,21 @@ public void EmptyProjectCanBeRead()
             Assert.That(packages, Is.Empty);
         }
 
+        [Test]
+        public void UnusualEncodingProjectCanBeRead()
+        {
+            const string NoProject =
+                @"<?xml version=""1.0"" encoding=""Windows-1252""?>
```
Comment (AnthonySteele) [inline NuKeeper.Inspection.Tests/RepositoryInspection/ProjectFileReaderTests.cs]: Is this a thing that you came across?

**Comment by AnthonySteele** [comment]
The exceptions look fine. 
I have never come across  `Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);` or `
System.Text.Encoding.CodePages` before.

File: NuKeeper.Inspection.Tests/RepositoryInspection/ProjectFileReaderTests.cs (hunk 1 above)
Comment (skolima) [reply NuKeeper.Inspection.Tests/RepositoryInspection/ProjectFileReaderTests.cs]: Comes from user issue #365 - but I should have used the VS classic project in the repro here, not the core SDK one.

Comment (skolima) [reply NuKeeper.Inspection.Tests/RepositoryInspection/ProjectFileReaderTests.cs]: Checked with a dotnet core project - technically this is a valid encoding and it works with dotnetcore sdk, so I'd keep this. Test case is valid, even if unusual.

**Pull request merged**